consensus/istanbul/core/types.go

```diff
@@ -134,10 +136,15 @@ func (m *message) FromPayload(b []byte, validateFn func([]byte, []byte) (common.
 			return err
 		}
 
-		_, err = validateFn(payload, m.Signature)
+		signerAdd, err := validateFn(payload, m.Signature)
+		if err != nil {
+			return err
+		}
+		if bytes.Compare(signerAdd.Bytes(), m.Address.Bytes()) != 0 {
```

Comment:
```suggestion
		if !bytes.Equal(signerAdd.Bytes(), m.Address.Bytes()) {
```
Isn't it better?

Reply:
It's from [Consensys/quorum@`0772362` (#853)](https://github.com/Consensys/quorum/pull/853/commits/0772362394b9f00bbd1d7368ce1f186bd9a19dd7).

Reply:
The latest code is same as Lewis's suggestion. It's good to be applied?

https://github.com/Consensys/quorum/blob/6cf0f5aa6d870b6d727d80a7b0d074c444369b81/consensus/istanbul/ibft/types/message.go#L94

Reply:
Reflected in a61842b7.